x-pack/plugins/transform/public/app/hooks/use_delete_transform.tsx

```diff
@@ -4,52 +4,273 @@
+          'xpack.transform.transformList.errorWithCheckingIfUserCanDeleteIndexNotificationErrorMessage',
+          {
+            defaultMessage: 'An error occurred checking if user can delete destination index',
+          }
+        )
+      );
+    }
+  }, [http, toastNotifications]);
+
+  useEffect(() => {
+    checkUserIndexPermission();
+
+    if (items.length === 1) {
+      const config = items[0].config;
+      const destinationIndex = Array.isArray(config.dest.index)
+        ? config.dest.index[0]
+        : config.dest.index;
+      checkIndexPatternExists(destinationIndex);
+    } else {
+      setIndexPatternExists(true);
+    }
+  }, [checkIndexPatternExists, checkUserIndexPermission, items]);
+
+  return {
+    userCanDeleteIndex,
+    deleteDestIndex,
+    indexPatternExists,
+    deleteIndexPattern,
+    toggleDeleteIndex,
+    toggleDeleteIndexPattern,
+  };
+};
 
 export const useDeleteTransforms = () => {
   const { overlays } = useAppDependencies();
   const toastNotifications = useToastNotifications();
   const api = useApi();
 
-  return async (transforms: TransformListRow[]) => {
+  return async (
+    transforms: TransformListRow[],
+    shouldDeleteDestIndex: boolean,
+    shouldDeleteDestIndexPattern: boolean
+  ) => {
     const transformsInfo: TransformEndpointRequest[] = transforms.map((tf) => ({
       id: tf.config.id,
       state: tf.stats.state,
     }));
 
     try {
-      const results: TransformEndpointResult = await api.deleteTransforms(transformsInfo);
+      const results: DeleteTransformEndpointResult = await api.deleteTransforms(
+        transformsInfo,
+        shouldDeleteDestIndex,
+        shouldDeleteDestIndexPattern
+      );
+      const isBulk = Object.keys(results).length > 1;
+      const successCount: Record<string, number> = {
+        transformJobDeleted: 0,
+        destIndexDeleted: 0,
+        destIndexPatternDeleted: 0,
+      };
       for (const transformId in results) {
         // hasOwnProperty check to ensure only properties on object itself, and not its prototypes
         if (results.hasOwnProperty(transformId)) {
-          if (results[transformId].success === true) {
-            toastNotifications.addSuccess(
-              i18n.translate('xpack.transform.transformList.deleteTransformSuccessMessage', {
-                defaultMessage: 'Request to delete transform {transformId} acknowledged.',
-                values: { transformId },
-              })
-            );
+          const status = results[transformId];
+          const destinationIndex = status.destinationIndex;
+
+          // if we are only deleting one modal, show the success toast messages
+          if (!isBulk) {
+            if (status.transformJobDeleted?.success) {
+              toastNotifications.addSuccess(
+                i18n.translate('xpack.transform.transformList.deleteTransformSuccessMessage', {
+                  defaultMessage: 'Request to delete transform job {transformId} acknowledged.',
```

Comment:
Please remove the word `job` - we just use the term `transform` - `Request to delete transform {transformId} acknowledged.`